doc/integrator/create_application.rst

```diff
@@ -513,3 +513,54 @@ Then run the make command for the user/instance you want to setup:
 
     ENV=<user> make -f <package>.mk user_mk
     make -f <user>_<instance>.mk build
+
+Dynamic configuration and autogenerated files
+---------------------------------------------
+
+Several files are autogenerated, their content depending of the variables you
+have set either in the main ``<package>.mk`` or a ``<user>.mk``
+
+The files can have either the extension ``.in`` or ``.mako``
+
+.in
+...
+
+If you use ``.in``, the variables are simply replaced and a file without the
+``.in`` extension is generated.
+
+**Syntax**
+
+In ``.in`` files, the variable replacement syntax is as follows::
+
+  ${<configsectionname>:<variablename>}
+
+where the ``<configsectionname>`` is the name of section in the configuration
+where the variable is defined,
```

Comment:
it was ..., now it's optional it can be specified only for backward compatibility ...